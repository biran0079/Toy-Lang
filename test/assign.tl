hello=1;
world=2;
toy=hello+world;
toy=toy*toy;
print(toy);
lang=toy*(hello+world);
print(lang);
print(lang=12345);
a=b=123;
print(a);
print(b);
