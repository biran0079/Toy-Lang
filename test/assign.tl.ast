[stmts, [=, [id, hello], [int, 1]], [=, [id, world], [int, 2]], [=, [id, toy], [+, [id, hello], [id, world]]], [=, [id, toy], [*, [id, toy], [id, toy]]], [call, [id, print], [exps, [id, toy]]], [=, [id, lang], [*, [id, toy], [+, [id, hello], [id, world]]]], [call, [id, print], [exps, [id, lang]]], [call, [id, print], [exps, [=, [id, lang], [int, 12345]]]], [=, [id, a], [=, [id, b], [int, 123]]], [call, [id, print], [exps, [id, a]]], [call, [id, print], [exps, [id, b]]]]
