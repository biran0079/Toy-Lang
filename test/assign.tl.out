9
27
12345
123
123
